pkg/proxy/util/utils.go

```diff
@@ -56,3 +72,54 @@ func ShouldSkipService(svcName types.NamespacedName, service *api.Service) bool
 	}
 	return false
 }
+
+// GetNodeAddresss list all match node IP addresses based on given cidr list.  Inject NetworkInterface for test purpose.
+// We expect the cidr list passed in is already validated.
+// Given `default-route`, it will the IP of host interface having default route.
+// Given `all-interfaces`, it will return `0.0.0.0/0` and `::/0` directly.
+func GetNodeAddress(cidrs []string, nw NetworkInterface) (sets.String, error) {
```

Comment:
Can this be smarter and return the minimum set?  Specifically, if any input is the zero CIDR, only rturn the zero CIDR for that address family.  E.g. [ "1.2.3.0/24", "0.0.0.0/0" ] should only return "0.0.0.0/0".  But if any v6 address was listed, that would still be included.

Secondly, it would be nice for the default input to be [] which means "everything in all families".

Reply:
> Can this be smarter and return the minimum set? 
Yes, we can. My algorithm is,

```
// First round of iteration to pick out `0.0.0.0/0` or `::/0` for the sake of excluding non-zero IPs.
for _, cidr := range cidrs {
	if IsZeroCIDR(cidr) {
		uniqueAddressList.Insert(cidr)
	}
}
// Second round of iteration to parse IPs based on cidr.
for _, cidr := range cidrs {
	if !uniqueAddressList.Has(zeroCIDR) {
		uniqueAddressList.Insert(cidr)
	}
}
```

> Secondly, it would be nice for the default input to be [] which means "everything in all families".

It's fixed now by removing the `all-interfaces` flag.